runtime/commonMain/src/kotlinx/serialization/Annotations.kt

```diff
@@ -10,60 +10,156 @@ import kotlinx.serialization.json.*
 import kotlin.reflect.*
 
 /**
- * Sets specific serializer for class, property or type argument.
- * When argument is omitted, plugin will generate default implementation inside the class.
+ * The main entry point to the serialization process.
+ * Applying [Serializable] to the Kotlin class instructs the serialization plugin to automatically generate [KSerializer]
+ * for the current class, that can be used to serialize and deserialize the class.
+ * The generated serializer can be accessed with `.serializer()` extension method that is generated by the plugin as well.
+ * ```
+ * @Serializable
+ * class MyData(val myData: AnotherData, val intProperty: Int, ...)
+ *
+ * // Produces JSON string using the generated serializer
+ * val jsonString = Json.encodeToJson(MyData.serializer(), instance)
+ * ```
+ *
+ * Additionally, the user-defined serializer can be specified using [with] parameter:
+ * ```
+ * @Serializable(with = MyAnotherDataCustomSerializer::class)
+ * class MyAnotherData(...)
+ *
+ * MyAnotherData.serializer() // <- returns MyAnotherDataCustomSerializer
+ * ```
+ *
+ * For annotated properties, specifying [with] parameter is mandatory and can be used to override
+ * serializer on the use-site without affecting the rest of the usages:
+ * ```
+ * @Serializable // By default is serialized as 3 byte components
+ * class RgbPixel(val red: Byte, val green: Byte, val blue: Byte)
```

Comment:
A common mistake that i've made in my talk once: RGB values are from 0 to 255, so `Byte` is not enough (unless you want to fiddle with sign)